### PR TITLE
fix(sveltekit): use `$env/static/private` in `framework-sveltekit`

### LINKED ISSUE #6439: Auth.js for SvelteKit doesn't work with Vercel Edge Functions
### Environment

npx envinfo --system --binaries --browsers --npmPackages "@sveltejs/kit,@sveltejs/adapter-vercel,@auth/core,@auth/sveltekit" 

  System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 72.55 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 19.3.0 - /opt/homebrew/bin/node
    Yarn: 1.22.18 - /opt/homebrew/bin/yarn
    npm: 9.2.0 - /opt/homebrew/bin/npm
  Browsers:
    Firefox Developer Edition: 110.0
    Safari: 16.2
  npmPackages:
    @auth/core: ^0.2.5 => 0.2.5 
    @auth/sveltekit: ^0.1.12 => 0.1.12 
    @sveltejs/adapter-vercel: ^1.0.5 => 1.0.5 
    @sveltejs/kit: ^1.0.0 => 1.1.1 

### Reproduction URL

https://github.com/evangelos-ch/sveltekit-auth-example

### Describe the issue

I'd like to use Vercel's Edge Functions, which SvelteKit does support as demonstrated in this example project: <https://vercel.com/templates/svelte/sveltekit-edge-functions>.

When I set up my project that way though and I also use Auth.js for Discord authentication, deployment breaks on Vercel, whereas it works just fine before and also just fine if I use `@sveltejs/adapter-auto` and do not set the edge flag.

The error from Auth.js is `Host must be trusted`, which breaks the entire site and not just any auth flow stuff. I tried to resolve this error with `AUTH_TRUST_HOST` or `NEXTAUTH_HOST` env variables to no avail. But after all I shouldn't need them on Vercel, right?

Then I tried deploying your SvelteKit example to Vercel and modifying it for Discord auth, and voila it worked just fine with the same variables as my other project. Then when I changed it over to Edge Functions, it broke with the same behaviour: Host must be trusted errors.

### How to reproduce

1. Deploy the project from my Reproduction URL to Vercel
2. Set the env variables `DISCORD_CLIENT_SECRET`, `DISCORD_CLIENT_ID` and `AUTH_SECRET` (or change them around to e.g. GitHub and then modify the appropriate files in the repo for another provider to test)
3. Redeploy since the build would've failed the first time due to the missing variables

### Expected behavior

The deployment should work, unless Edge incompatibility is some intended or unavoidable limitation of Auth.js, which should then be documented somewhere.

### PR DESCRIPTION
**Pull request by ThangHuuVu**
> _NOTE_:
>
> - It's a good idea to open an issue first to discuss potential changes.
> - Please make sure that you are _NOT_ opening a PR to fix a potential security vulnerability. Instead, please follow the [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md) to disclose the issue to us confidentially.

## ☕️ Reasoning

Make our environment variables available in the build step as static so they can be used in the edge functions. 

## 🧢 Checklist

- [ ] Documentation
- [x] Tests
- [x] Ready to be merged

## 🎫 Affected issues

Fixes: #6439 

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/frameworks-sveltekit/.env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# These environment variables are used by Sveltekit to inject into `$env/static/private`
```
Comment (balazsorban44) [inline packages/frameworks-sveltekit/.env:1]: Why is this file added to the package itself?

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Hmm. 😬 This would contradict #6231.

The only way I could see this get solved is if we add an `env` config option to `SvelteKitAuthConfig` and there you can pass all the necessary variables. This would require an extra step from the user though.

File: packages/frameworks-sveltekit/.env (hunk 1 above)
Comment (ThangHuuVu) [reply packages/frameworks-sveltekit/.env]: Sorry, I used TS `declare` instead 😅

**Comment by ThangHuuVu** [comment]
Closing in favor of the upstream fix linked in the issue
